mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -233,7 +232,10 @@ object KMeansModel extends MLReadable[KMeansModel] {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
 
-      val clusterCenters = if (majorVersion(metadata.sparkVersion) >= 2) {
+      val versionRegex = "([0-9]+)\\.(.+)".r
```

Comment:
Your change is reverting some other changes like this one. Can you clean this up to only reflect the changes you intend?